execute/observation.go

```diff
@@ -183,15 +227,21 @@ func readAllMessages(
 
 		ranges, err := computeRanges(reports)
 		if err != nil {
-			return exectypes.MessageObservations{}, err
+			lggr.Errorw("unable to compute ranges", "err", err)
+			continue
 		}
 
 		// Read messages for each range.
 		for _, seqRange := range ranges {
 			// TODO: check if srcChain is supported.
 			msgs, err := ccipReader.MsgsBetweenSeqNums(ctx, srcChain, seqRange)
 			if err != nil {
-				return exectypes.MessageObservations{}, err
+				lggr.Errorw("unable to read messages",
```

Comment:
👍

Reply:
This error message could also correlate this to the merkle roots which we can't process, which could be helpful for debugging. Getting the roots would also be part of the suggestion above to combine `intersectCommitReportsAndMessages` with this function.